### PR TITLE
haxe 4.2.2

### PR DESCRIPTION
**Pull request by chenrui333**
Created with `brew bump-formula-pr`.

~`resource` blocks may require updates.~

### CODE REVIEW

**Comment by cho-m** [comment]
```make
The following dependencies couldn't be met:
  - haxe → camlp5 = 8.00~alpha05 → ocaml < 4.12.0
      base of this switch (use `--unlock-base' to force)
```

Can consider either bumping dependency: HaxeFoundation/haxe#10243
Or using `opam` to switch to older ocaml compiler

**Comment by cho-m** [comment]
@nandahkrishna I think `opam` may be ignoring uncommitted changes, which could be why patch failed.

**Comment by nandahkrishna** [comment]
I've done the easier thing and used the older `ocaml`, if anyone is able to update the dependency instead (and if that's a better option), feel free to overwrite my commit.

**Comment by cho-m** [comment]
I think the Intel Big Sur node might fail since it was upgraded to macOS 11.3.
Last formula that used 4.10.2 OCaml compiler ended hitting `<version>` header issue: https://github.com/Homebrew/homebrew-core/pull/76176#issuecomment-830463866

File: Formula/haxe.rb [hunk 1]
```diff
@@ -57,6 +57,10 @@ def install
       ENV["OPAMYES"] = "1"
       ENV["ADD_REVISION"] = "1" if build.head?
       system "opam", "init", "--no-setup", "--disable-sandboxing"
+
+      # Remove when the camlp5 dependency is updated to 8.0.0 or later
```
Comment (carlocab) [inline Formula/haxe.rb:61]: Brewed `camlp5` is 8.0.0. Can we use that?

Comment (nandahkrishna) [reply Formula/haxe.rb]: Missed this, but I got it to work locally by applying the upstream patch and using `--working-dir`, as pointed out by @cho-m.

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**